src/Ocelot/DownstreamUrlCreator/Middleware/DownstreamUrlCreatorMiddleware.cs

```diff
@@ -1,156 +1,179 @@
+            {
+                //todo make sure this works, hopefully there is a test ;E
+                httpContext.Items.DownstreamRequest().Scheme = downstreamRoute.DownstreamScheme;
+            }
+
+            var internalConfiguration = httpContext.Items.IInternalConfiguration();
+
+            if (ServiceFabricRequest(internalConfiguration, downstreamRoute))
+            {
+                var pathAndQuery = CreateServiceFabricUri(downstreamRequest, downstreamRoute, templatePlaceholderNameAndValues, response);
+
+                //todo check this works again hope there is a test..
+                downstreamRequest.AbsolutePath = pathAndQuery.Path;
+                downstreamRequest.Query = pathAndQuery.Query;
+            }
+            else
+            {
+                var dsPath = response.Data;
+
+                if (ContainsQueryString(dsPath))
+                {
+                    downstreamRequest.AbsolutePath = GetPath(dsPath);
+
+                    if (string.IsNullOrEmpty(downstreamRequest.Query))
+                    {
+                        downstreamRequest.Query = GetQueryString(dsPath);
+                    }
+                    else
+                    {
+                        //downstreamRequest.Query += GetQueryString(dsPath).Replace('?', '&');
+                        var newQueryString = GetQueryString(dsPath).Replace('?', '&');
+                        newQueryString = MergeQueryStringsWithoutDuplicateValues(downstreamRequest.Query, newQueryString);
+                        downstreamRequest.Query = "?" + newQueryString;                    }
+                }
+                else
+                {
+                    RemoveQueryStringParametersThatHaveBeenUsedInTemplate(downstreamRequest, templatePlaceholderNameAndValues);
+
+                    downstreamRequest.AbsolutePath = dsPath.Value;
+                }
+            }
+
+            Logger.LogDebug($"Downstream url is {downstreamRequest}");
+
+            await _next.Invoke(httpContext);
+        }
+
+        private static string MergeQueryStringsWithoutDuplicateValues(string queryString, string newQueryString)
+        {
+            var queries = HttpUtility.ParseQueryString(queryString);
+            var newQueries = HttpUtility.ParseQueryString(newQueryString);
+
+            var dict = new Dictionary<string, string>();
+            foreach (var key in newQueries.AllKeys)
+            {
+                if (!string.IsNullOrEmpty(key))
+                {
+                    dict.Add(key, newQueries[key]);
+                }
+            }
+
+            foreach (var key in queries.AllKeys)
+            {
+                if (string.IsNullOrEmpty(key) || dict.ContainsValue(queries[key]))
+                {
+                    continue;
+                }
+
+                dict.Add(key, queries[key]);
+            }
+
+            return string.Join("&", dict.Select(kvp => string.Format("{0}={1}", kvp.Key, kvp.Value)));
```

Comment:
```suggestion
            return string.Join('&', dict.Select(kvp => $"{kvp.Key}={kvp.Value}"));
```

Reply:
Fixed in https://github.com/ThreeMammals/Ocelot/pull/1182/commits/275d034250762557f50e8461d27b3446893b54a1